### PR TITLE
Descriptor & mudmail cleanups

### PR DESCRIPTION
**Pull request by rmsr**
Many of the Descriptor member names are too short and/or misnamed. Clean up that technical debt.

Mudmail changes:
* prevent mudmailing nonexistent people from mail cmd
* disable mudmailing zero talens (because you'd still pay the sending-talens rate)
* remove no-mudmail config

### CODE REVIEW

File: code/code/sys/connect.cc [hunk 1]
```diff
@@ -83,6 +83,7 @@ tot_pages(0),
 cur_page(0),
 str(NULL),
 max_str(0),
+mail_talens(0),
```
Comment (cizra) [inline code/code/sys/connect.cc]: All these copy and assignment constructors should be deleted

File: code/code/sys/connect.h [hunk 2]
```diff
@@ -491,9 +491,6 @@ class Descriptor
     std::string mudclient;
     std::string clientversion;
 
-    // Functions
-  private:
-    Descriptor();  // prevent default constructor from being used
```
Comment (cizra) [inline code/code/sys/connect.h]: `Descriptor() = delete;`

File: code/code/misc/being.h [hunk 3]
```diff
@@ -567,7 +567,7 @@ class TBeing : public TThing {
     int missVictim(TBeing *, TThing *, spellNumT);
 
     // Postmaster
-    void postmasterSendMail(const char *, TMonster *);
+    void postmasterSendMail(sstring, TMonster *);
```
Comment (cizra) [inline code/code/misc/being.h]: `const sstring&`

**Comment by rmsr** [comment]
uuugh I managed to rebase myself into a corner on this and #139, closing them both and opening a new unified PR

**Comment by rmsr** [comment]
hahaha sucker I guess I'm not making a new PR but just re-opening this one. The new commits start at ff54b4b so all the old reviewing is still valid.

File: code/code/misc/being.h (hunk 3 above)
Comment (rmsr) [reply code/code/misc/being.h]: fixed along with another constification

File: code/code/misc/create_objs.cc [hunk 4]
```diff
@@ -1096,8 +1096,8 @@ static void change_obj_short_desc(TBeing *ch, TObj *o, editorEnterTypeT type)
   ch->sendTo("(Terminate with a ~ on the SAME LINE. Press <ENTER> again to continue)\n\r");
   o->shortDescr = "";
   ch->desc->str = &o->shortDescr;
-//  ch->desc->max_str = MAX_STRING_LENGTH;
-  ch->desc->max_str = MAX_NAME_LENGTH-1;
+//  ch->desc->edit_str_maxlen = MAX_STRING_LENGTH;
```
[the remark this answers is not in the document]
Comment (rmsr) [reply code/code/misc/create_objs.cc]: removed

File: code/code/sys/connect.cc [hunk 5]
```diff
@@ -82,7 +82,7 @@ showstr_head(NULL),
 tot_pages(0),
 cur_page(0),
 str(NULL),
-max_str(0),
+edit_str_maxlen(0),
```
[the remark this answers is not in the document]
Comment (rmsr) [reply code/code/sys/connect.cc]: I dunno, apparently we don't need most of them, but I'm not going to try to figure out which ones are important right now. Better done as part of a more general ctor cleanup.

File: code/code/sys/connect.h [hunk 6]
```diff
@@ -496,8 +496,6 @@ class Descriptor
     Descriptor();  // prevent default constructor from being used
   public:
     Descriptor(TSocket *);
-    Descriptor(const Descriptor &);
-    Descriptor & operator=(const Descriptor &a);
```
[the remark this answers is not in the document]
Comment (rmsr) [reply code/code/sys/connect.h]: Well masking them like you say doesn't result in any compile failures. No, I don't think you should be able to copy a descriptor, that sounds like instant fail.

File: code/code/sys/connect.h (hunk 2 above)
Comment (rmsr) [reply code/code/sys/connect.h]: masked

**Comment by rmsr** [comment]
OK, should be ready for a new review.

**Comment by wershlak** [comment]
@cizra Should we close this or do you think there's value in working out all the conflicts? I'm guessing @rmsr isn't going to finish it.

**Comment by drewhershey** [comment]
Gonna close this out since at this point it has a million merge conflicts and would make sense to just start fresh anyway, if someone were going to tackle similar changes.

**Pull request closed without merging**